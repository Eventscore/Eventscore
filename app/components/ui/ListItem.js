//<ListItem title={'Hello World'} />
import React, { Component, PropTypes } from 'react';
import { ListItem } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppStyles } from '@theme/';

class CustomListItem extends Component {
  static propTypes = {
    containerStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    titleStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    subtitleStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
  }

  static defaultProps = {
    containerStyle: [],
    titleStyle: [],
    subtitleStyle: [],
  }

  listItemProps = () => {
    // Defaults
    const props = {
      title: 'Coming Soon...',
      chevronColor: AppColors.textSecondary,
      underlayColor: AppColors.border,
      ...this.props,
      containerStyle: [{
        backgroundColor: AppColors.listItemBackground,
        borderTopColor: AppColors.border,
        borderBottomColor: AppColors.border,
      }],
      titleStyle: [AppStyles.baseText],
      subtitleStyle: [AppStyles.subtext],
    };

    if (this.props.containerStyle) {
      props.containerStyle.push(this.props.containerStyle);
    }

    if (this.props.titleStyle) {
      props.titleStyle.push(this.props.titleStyle);
    }

    if (this.props.subtitleStyle) {
      props.subtitleStyle.push(this.props.subtitleStyle);
    }

    return props;
  }

  render = () => <ListItem {...this.listItemProps()} />;
}

export default CustomListItem;
