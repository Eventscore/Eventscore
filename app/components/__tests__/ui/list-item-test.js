/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ListItem from '@ui/ListItem';

it('ListItem renders correctly', () => {
  const tree = renderer.create(
    <ListItem
      title={'Hello world'}
      subTitle={'Sub title'}
      roundAvatar
      avatar={require('../../../image.png')}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
