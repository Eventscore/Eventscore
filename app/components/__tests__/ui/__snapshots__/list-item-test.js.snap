// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<View
  onLongPress={undefined}
  onPress={undefined}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#ededed",
        "borderBottomWidth": 1,
        "paddingBottom": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#D0D1D5",
          "borderTopColor": "#D0D1D5",
        },
        Array [],
      ],
    ]
  }
  underlayColor="#D0D1D5"
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginLeft": 10,
        },
        undefined,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 34,
            "width": 34,
          },
          Object {
            "borderRadius": 17,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontSize": 17.5,
                },
                Object {
                  "marginLeft": 10,
                },
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                  Array [],
                ],
                undefined,
              ],
            ]
          }
        >
          Hello world
        </Text>
      </View>
      <View
        style={undefined}
      >
        <View />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.15,
          "justifyContent": "center",
        }
      }
    >
      <View
        onLongPress={undefined}
        onPress={undefined}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 28,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {},
                  undefined,
                ],
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          î—Œ
        </Text>
      </View>
    </View>
  </View>
</View>
`;
